bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/config/TestBigtableVaneerSettingsFactory.java

```diff
@@ -0,0 +1,414 @@
+      for (Row outputRow : rowStream) {
+
+        //Checking if the received output's KEY is same as above.
+        ByteString key = outputRow.getKey();
+        LOG.info("found key: " + key.toStringUtf8());
+        assertEquals(TEST_KEY, outputRow.getKey());
+
+        for (RowCell cell : outputRow.getCells()) {
+          //Checking if the received output is KEY sent above.
+          ByteString value = cell.getValue();
+          LOG.info("Value found: " + value.toStringUtf8());
+          assertEquals(TEST_VALUE, value);
+        }
+      }
+
+      //Removing the table.
+      adminClient.deleteTable(TABLE_ID);
+    } finally {
+      //Removing Table in case of some exception occurres.
+      boolean tableExist = adminClient.exists(TABLE_ID);
+      if (tableExist) {
+        adminClient.deleteTable(TABLE_ID);
+      }
+      assertFalse(adminClient.exists(TABLE_ID));
+    }
+  }
+
+  @Test
+  public void testProjectIdIsRequired() throws IOException {
+    BigtableOptions options = BigtableOptions.builder().build();
+
+    expectException.expect(IllegalStateException.class);
+    expectException.expectMessage("Project ID is required");
+    dataSettings = BigtableVaneerSettingsFactory.createBigtableDataSettings(options);
+  }
+
+  @Test
+  public void testInstanceIdIsRequired() throws IOException {
+    BigtableOptions options = BigtableOptions.builder().setProjectId(TEST_PROJECT_ID).build();
+
+    expectException.expect(IllegalStateException.class);
+    expectException.expectMessage("Instance ID is required");
+    dataSettings = BigtableVaneerSettingsFactory.createBigtableDataSettings(options);
+  }
+
+  @Test
+  public void testWhenRetriesAreDisabled() throws IOException {
+    RetryOptions retryOptions = RetryOptions.builder().setEnableRetries(false).build();
+    BigtableOptions options =
+        BigtableOptions.builder()
+            .setProjectId(TEST_PROJECT_ID).setInstanceId(TEST_INSTANCE_ID)
+            .setRetryOptions(retryOptions).build();
+
+    expectException.expect(IllegalStateException.class);
+    expectException.expectMessage("Disabling retries is not currently supported.");
+    dataSettings = BigtableVaneerSettingsFactory.createBigtableDataSettings(options);
+  }
+
+  @Test
+  public void testWithNullCredentials() throws IOException {
+    BigtableOptions options =
+        BigtableOptions.builder()
+            .setProjectId(TEST_PROJECT_ID).setInstanceId(TEST_INSTANCE_ID)
+            .setCredentialOptions(CredentialOptions.nullCredential())
+            .setUserAgent(TEST_USER_AGENT).build();
+    dataSettings = BigtableVaneerSettingsFactory.createBigtableDataSettings(options);
+    assertTrue(dataSettings.getCredentialsProvider() instanceof NoCredentialsProvider);
+  }
+
+  @Test
+  public void testRetriableRpcs() throws IOException {
+    dataSettings = BigtableVaneerSettingsFactory.createBigtableDataSettings(bigtableOptions);
+
+    //Verifying RetrySettings & RetryCodes of retryables methods.
```

Comment:
Unless you have a method named retryables, the sentence should be: //Verifying RetrySettings & RetryCodes of retryable methods.